### ISSUE TITLE
Some confusion with media statement

### ISSUE THREAD

**Issue by gazley** [ausi/respimagelint#12]
```
<picture>
    <!--[if IE 9]><video style="display: none;"><![endif]-->
                                        <source media="(min-width: 960px) and (-webkit-min-device-pixel-ratio: 1.25), (min-width: 960px) and (min-resolution: 120dpi)" sizes="calc(33vw - 25px)" srcset="/site/assets/files/1176/stephen_micheal15.706x0.jpg 353w">
                                            <source media="(min-width: 960px)" sizes="calc(33vw - 25px)" srcset="/site/assets/files/1176/stephen_micheal15.353x0.jpg 353w">
        <!--[if IE 9]></video><![endif]-->
    <img sizes="100vw" srcset="/site/assets/files/1176/stephen_micheal15.430x0.jpg 430w,/site/assets/files/1176/stephen_micheal15.909x0.jpg 909w,/site/assets/files/1176/stephen_micheal15.860x0.jpg 860w,/site/assets/files/1176/stephen_micheal15.1818x0.jpg 1818w" src="/site/assets/files/1176/stephen_micheal15.909x0.jpg" alt=""></picture>
```

With code like the above, the linter is reporting:

```Descriptor 353w doesn’t match the image size of 706x706 from /site/assets/files/1176/stephen_micheal15.706x0.jpg.```

I accept this as a fact and I may change my code to suit. However, that image would only display on a retina 2x screen where the image size is still rendered at 353px wide. So, semantically, it's the same width being satisfied by an image exactly twice the size. For the sake of correctness, I could specify 706w and it wouldn't break anything and be technically correct. However, the report goes on to say ...

```At a viewport width of 1000 the image was displayed 293 pixels wide, but the closest provided image has a width of 706 which is 58% off. The affected viewport widths are 300-400, 960-1210.```

On my non-retina iMac 27", the browser is receiving:

http://localhost:8888/site/assets/files/1176/stephen_micheal15.353x0.jpg

It seems that in this circumstance, the linter is possibly being confused up by the retina source media query, suggesting that a 706 pixel wide image is being used but being displayed at 293 pixels wide in the browser, hence a large image in for a small rendering area. However, this is only the case when a retina device is in use and which case, the 706 pixel image would be appropriate.

I'm not saying this is a bug in the linter but, its reporting of this issue isn't totally correct. Do you have any thoughts around this issue?

Many thanks!

**Comment by ausi** [comment]
It looks like there is an issue with parsing the `media` attribute, the linter uses the wrong `<source>` element while looking for a fitting image.

Specifying a non-matching `w` descriptor in a `srcset` is wrong and has an effect if you don’t resize your image via CSS.

Your example code seems to have issues, but it’s hard to tell without a live webpage. Can you provide a link to a webpage with that image on it?

**Comment by ausi** [comment]
Media query parsing should be more robust now and the *„missing fitting image source“* errors should no longer appear.
